diff --git a/ui/gfx/linux/gbm_wrapper.cc b/ui/gfx/linux/gbm_wrapper.cc
index 1b5d52cbb3..5283e3dede 100644
--- a/ui/gfx/linux/gbm_wrapper.cc
+++ b/ui/gfx/linux/gbm_wrapper.cc
@@ -36,6 +36,12 @@
 #include "base/strings/stringize_macros.h"
 #endif
 
+extern "C" {
+// Weak import of the new GBM API.
+int gbm_bo_get_fd_for_plane(struct gbm_bo* bo, int plane)
+    __attribute__((weak_import));
+}
+
 namespace ui {
 namespace gbm_wrapper {
 namespace {
@@ -60,6 +66,11 @@ base::ScopedFD GetPlaneFdForBo(gbm_bo* bo, size_t plane) {
 #if defined(MINIGBM)
   return base::ScopedFD(gbm_bo_get_plane_fd(bo, plane));
 #else
+  // gbm has fd per plane support
+  if (gbm_bo_get_fd_for_plane) {
+    return base::ScopedFD(gbm_bo_get_fd_for_plane(bo, static_cast<int>(plane)));
+  }
+
   const int plane_count = GetPlaneCount(bo);
   DCHECK(plane_count > 0 && plane < static_cast<size_t>(plane_count));
 
